### ISSUE TITLE
[Data] Many Options Strikes are missing

### ISSUE THREAD

**Issue by xunamcapital** [iexg/IEX-API#1738]
### Please keep all posted issues professional and constructive. We may delete posts that do not provide sufficient information, are outright negative and unproductive, or are not directly related to IEX Cloud.

**Email account associated with your IEX Account**
roman@xunam.com

**Date/time of the issue**
Any time

**What you did**
https://cloud.iexapis.com/stable/stock/market/options/20210820?token=token&symbols=o

**What you expected to happen**
Get information on every strike.

**What actually happened**
Missing strikes from the returned data set - easily confirmed since many strikes are skipped, while strikes below and above exist.
For ticker O for example strike for price 50 should be active, but it is missing.

**API Url (if applicable)**
https://cloud.iexapis.com/stable/stock/market/options/20210820?token=token&symbols=o

**Additional relevant information**
The strikes are missing for almost any symbols.

**Comment by timkpaine** [comment]
@xunamcapital have you tried using the new options refdata endpoints?
`https://cloud.iexapis.com/stable/ref-data/options/symbols/O?token=pk_...`

**Comment by timkpaine** [comment]
I will make a note to add another path parameter to filter by expiry to that and the futures symbols endpoints

**Comment by xunamcapital** [comment]
@timkpaine thank you for the API suggestion - it could be useful in getting all reference data, but it does not provide any last Trade or bid ask prices. We are looking for either last trade or bed ask prices from previous day - that are provided by 
 https://cloud.iexapis.com/stable/stock/market/options/20210820?token=token&symbols=o  (All data fields are perfect for us, it just seems to be missing data).

**Comment by timkpaine** [comment]
You can query the new chart endpoint for that info. But generally if it's missing from the stock endpoint there wasn't a trade.

**Comment by xunamcapital** [comment]
Thank you @timkpaine ! We will try the new endpoint

**Comment by timpaine-iex** [comment]
@xunamcapital going to close this, the idea behind the new endpoints are threefold:

- First, the original endpoint was built off trade data, so if something didnt trade it would be missing from that endpoint
- Second, this endpoint is weird. theres no way to get "options symbol" alone, and we mixed reference data with market data. with the new reference data endpoints its possible to pull reference data separate from market data like we do for stocks and futures (which ends up being cheaper in number of credits)
- It was a bit confusing to query for options historical EOD data since it didnt really match the pattern for stocks. Now it matches (query refdata to get symbols, query /options to get historical data) and we've brought it in line with the time-series APIs

I've kept these endpoints as "beta" for now so we can continue to make improvements. Theres a bunch of stuff in the pipeline: we already consume international options data from like 30+ exchanges, it just hasnt been hooked in yet, we also have plans for realtime options data (probably OPRA only to start). Feel free to write in with suggestions, I read everything that gets submitted to http://iexcloud.io/console/roadmap
